core/domain/exp_domain.py

```diff
@@ -2964,6 +2964,46 @@ def delete_state(self, state_name):
 
         del self.states[state_name]
 
+    def get_trainable_states_dict(self, old_states):
+        """Retrieves the state names of all trainable states in an exploration
+        segregated into state names with changed and unchanged answer groups.
+
+        Args:
+            old_states: dict. List of State Domain objects.
+
+        Returns:
+            dict. The trainable states dict.
+        """
+        trainable_states_dict = {
```

Comment:
Hm, I guess I withdraw my objection to the dict :) I can't think of a cleaner way to do it.